agent/consul/internal_endpoint_test.go

```diff
@@ -1897,6 +1898,66 @@ func TestInternal_ServiceTopology(t *testing.T) {
 	})
 }
 
+func TestInternal_ServiceTopology_RoutingConfig(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	t.Parallel()
+	dir1, s1 := testServer(t)
+	defer os.RemoveAll(dir1)
+	defer s1.Shutdown()
+
+	testrpc.WaitForLeader(t, s1.RPC, "dc1")
+
+	codec := rpcClient(t, s1)
+	defer codec.Close()
+
+	// dashboard -> routing-config -> { counting, counting-v2 }
+	registerTestRoutingConfigTopologyEntries(t, codec)
+
+	t.Run("dashboard", func(t *testing.T) {
+		retry.Run(t, func(r *retry.R) {
+			args := structs.ServiceSpecificRequest{
+				Datacenter:  "dc1",
+				ServiceName: "dashboard",
+			}
+			var out structs.IndexedServiceTopology
+			require.NoError(r, msgpackrpc.CallWithCodec(codec, "Internal.ServiceTopology", &args, &out))
+			require.False(r, out.FilteredByACLs)
+			require.Equal(r, "http", out.ServiceTopology.MetricsProtocol)
+
+			require.Empty(r, out.ServiceTopology.Downstreams)
+
+			expectDown := map[string]structs.IntentionDecisionSummary{
+				"counting":    {DefaultAllow: true, Allowed: true},
```

Comment:
These downstream decision summaries and sources should probably be excluded as well, since I don't believe either of them are transparent proxies

Reply:
Ah I misunderstood your last comment about downstreams. I added the `if hasTransparent` check to downstream intentions as well